cachedcarthage/defaultstateprovider.go

```diff
@@ -0,0 +1,96 @@
+package cachedcarthage
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/pathutil"
+)
+
+// DefaultStateProvider reads the current state of a cached Carthage project.
+type DefaultStateProvider struct {
+}
+
+// ParseState ...
+func (provider DefaultStateProvider) ParseState(project Project) (ProjectState, error) {
+	buildDirExists, buildDirFiles := provider.parseBuildDirecrtoyState(project.buildDir())
+	cacheFileExists, cacheFileContent, err := provider.parseCacheFileState(project.cacheFilePath())
+	if err != nil {
+		return ProjectState{}, err
+	}
+	resolvedFileExists, resolvedFileContent, err := provider.parseResolvedFileState(project.resolvedFilePath())
+	if err != nil {
+		return ProjectState{}, err
+	}
+
+	carthageDirExists, err := pathutil.IsPathExists(project.carthageDir())
+	if err != nil {
+		return ProjectState{}, fmt.Errorf("failed to check if dir exists at (%s), error: %s", project.carthageDir(), err)
+	}
+
+	return ProjectState{
+		buildDirExists: buildDirExists,
+		buildDirFiles:  buildDirFiles,
```

Comment:
Is this used anywhere?

Reply:
Wow, nice catch. It should have actually been used in cache intact check. I will add the usage and a test for it.

Reply:
I have rather merged the two values into `buildDirNotEmpty` since we do not actually need the two separate, just that if the build dir is empty.